src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +35,15 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                byte[] payloadLengthBuffer = new byte[sizeof(long)]; // Stream's Position is a long
```

Comment:
We should talk about whether we want to add a more elaborate header here.

Reply:
Probably at most a version field and then the payload length. The payload can be differentiated by version if necessary.

Reply:
Was there other info you wanted to include in it? The primary purpose of the header is a workaround for the buffer issue with `Console.ReadLine` as opposed to passing data to the extension, but I'm curious what you had in mind.

Reply:
Let's write out:
- Version (4-byte integer); its value is fixed to 1
- Payload length in bytes (8-byte integer)
- Payload as bytes

The version will indicate the format of the remainder of the data stream. Egress extensions should validate the version is what is expected and otherwise fail (throw exception).

We likely won't change the remainder of the stream in any significant manner after 8.0 is released, but this allows us to have that ability in the future if necessary.